avocado/plugins/sysinfo.py

```diff
@@ -36,13 +37,13 @@ def _init_sysinfo(self, job_logdir):
             self.sysinfo = sysinfo.SysInfo(basedir=basedir)
 
     def pre(self, job):
-        if job.config.get('sysinfo', None) != 'on':
+        if not settings.get('sysinfo.collect', 'enabled', bool):
             return
         self._init_sysinfo(job.logdir)
         self.sysinfo.start_job_hook()
 
     def post(self, job):
-        if job.config.get('sysinfo', None) != 'on':
+        if not settings.get('sysinfo.collect', 'enabled', bool):
```

Comment:
Also, keep in mind that we could very well have more than one job at a time, each with a different configuration.

Reply:
See if my previous comment makes sense.